include/experimental/__p1673_bits/scaled.hpp

```diff
@@ -50,87 +50,78 @@ namespace experimental {
 inline namespace __p1673_version_0 {
 namespace linalg {
 
-template<class Reference, class ScalingFactor>
-class scaled_scalar {
-private:
-  Reference value;
-  const ScalingFactor scaling_factor;
-
-  using result_type = decltype (value * scaling_factor);
-public:
-  scaled_scalar(Reference v, const ScalingFactor& s) :
-    value(v), scaling_factor(s) {}
-
-  operator result_type() const { return value * scaling_factor; }
-
-  // mfh 03 Sep 2019: This doesn't appear to be necessary.
-
-  // template<class T>
-  // decltype(auto) operator+ (T update) const {
-  //   return value * scaling_factor + update;
-  // }
-};
-
-template<class Accessor, class S>
+template<class ScalingFactor, class Accessor>
 class accessor_scaled {
 public:
-  using element_type  = typename Accessor::element_type;
+  using reference     =
+    scaled_scalar<ScalingFactor, typename Accessor::reference,
+      std::remove_cv_t<typename Accessor::element_type>>;
+  using element_type  = std::add_const_t<typename reference::value_type>;
   using pointer       = typename Accessor::pointer;
-  using reference     = scaled_scalar<typename Accessor::reference, S>;
-  using offset_policy = accessor_scaled<typename Accessor::offset_policy, S>;
-
-  accessor_scaled() = default;
-
-  accessor_scaled(Accessor a, S sval) :
-    acc_(a), scale_factor_(sval) {}
-
+  using offset_policy =
+    accessor_scaled<ScalingFactor, typename Accessor::offset_policy>;
+
+  accessor_scaled(ScalingFactor scaling_factor, Accessor accessor) :
```

Comment:
It would be good to keep the default constructor - without it something in my code is broken.

Reply:
Please see note above; thanks!

Reply:
Update: it was possible to work around without changing the spec or reference implementation, yay!